### ISSUE TITLE
Define and uniformly implement a typeSpec filter on composite methods as appropriate

### ISSUE THREAD

**Issue by ntouran** [terrapower/armi#129]
In #127, @youngmit asked if we should add a Flags/typeSpec filter to the composite method that is being applied as the replacement for the outgoing `getEnrichment` block method. This is a good point and the answer is: most likely. 

* Methods on the composite tree that scan over children should offer some consistent and uniform method for filtering the children (e.g. the flags mechanism)

* We need a consistent definition of how the typespec would be applied. If you ask a core for fuel and it has fuel components in blocks that are marked "heat exchanger", would those pass the filter? this must be defined and documented. 

* which methods should we apply this kind of filtration to?

* Other issues?

**Comment by ntouran** [comment]
Does anyone like the idea of embedding the `exact=True` metadata directly into the Flags? as like a special flag flag `Flags.EXACT`?

**Comment by onufer** [comment]
Yeah I like that a lot. No longer being slightly uncertain about what the default is for a specific method.

**Comment by youngmit** [comment]
Exact should not be a flag, but it would make a lot of sense to ship flags, plus exactness, and perhaps even depth semantics into a TypeSpec class of some sort. This would make consistent method signatures easier to maintain, and make it easy to document how filtering works
